theseus/extlib/baspacho_solver.cpp

```diff
@@ -175,13 +175,23 @@ void NumericDecomposition::add_MtM(const torch::Tensor& val,
     }
 }
 
-void NumericDecomposition::damp(double alpha, double beta) {
+void NumericDecomposition::damp(const torch::Tensor& alpha, const torch::Tensor& beta) {
+    TORCH_CHECK(alpha.dtype() == torch::kFloat64);
+    TORCH_CHECK(alpha.dim() == 1);
+    TORCH_CHECK(alpha.size(0) == data.size(0));
+    TORCH_CHECK(beta.dtype() == torch::kFloat64);
+    TORCH_CHECK(beta.dim() == 1);
+    TORCH_CHECK(beta.size(0) == data.size(0));
+
 #ifdef THESEUS_HAVE_CUDA
     if (dec->isCuda) {
-        damp_cuda(alpha, beta);
+        TORCH_CHECK(alpha.device().is_cuda());
+        TORCH_CHECK(beta.device().is_cuda());
+        damp_cuda(alpha.data_ptr<double>(), beta.data_ptr<double>());
         return;
     }
 #endif
+    TORCH_CHECK(alpha.device().is_cpu());
```

Comment:
No similar beta cpu check?

Reply:
Added some macros to clean up all the tensor checks. See 919b8783b030a0be89e4033966b855440482c102.